controller/src/controller.rs

```diff
@@ -203,3 +205,138 @@ impl<T: BottlerocketShadowClient> BrupopController<T> {
         }
     }
 }
+
+// Get node and BottlerocketShadow names
+#[instrument]
+fn get_associated_bottlerocketshadow_name() -> Result<String> {
+    let associated_node_name = env::var("MY_NODE_NAME").context(error::GetNodeName)?;
+    let associated_bottlerocketshadow_name = brs_name_from_node_name(&associated_node_name);
+
+    event!(
+        Level::INFO,
+        ?associated_bottlerocketshadow_name,
+        "Found associated bottlerocketshadow name."
+    );
+
+    Ok(associated_bottlerocketshadow_name)
+}
+
+/// sort shadows list which uses to determine the order of node update
+/// logic1: sort shadows by crash count
+/// logic2: move the shadow which associated bottlerocketshadow node hosts controller pod to the last
+#[instrument(skip())]
+fn sort_shadows(
+    mut shadows: Vec<BottlerocketShadow>,
+    associated_brs_name: &str,
+) -> Result<Vec<BottlerocketShadow>> {
+    let mut associated_brs_node_position: usize = usize::MAX;
+    // sort shadows by crash count
+    shadows.sort_by(|a, b| a.compare_crash_count(b));
+
+    // move the shadow which associated bottlerocketshadow node hosts controller pod to the last
+    for (position, brs) in shadows.iter().enumerate() {
+        if brs
+            .metadata
+            .name
+            .as_ref()
+            .context(error::MissingBottlerocketShadowName)?
+            == associated_brs_name
+        {
+            associated_brs_node_position = position;
+            break;
+        }
+    }
+
+    // if it doesn't find the brs, it means those brss aren't ready and the program should skip sort.
+    if associated_brs_node_position != usize::MAX {
+        let last_brs = shadows[associated_brs_node_position].clone();
+        shadows.remove(associated_brs_node_position.clone());
+        shadows.push(last_brs.clone());
```

Comment:
I don't think a clone here is necessary, since `last_brs` was cloned out of the vector and ownership is being pushed back into it here.